contracts/validation/validation.go

```diff
@@ -5,10 +5,16 @@ type Option func(map[string]any)
 type Validation interface {
 	// Make create a new validator instance.
 	Make(data any, rules map[string]string, options ...Option) (Validator, error)
+	// AddFilter add the custom filter.
+	AddFilter(name string, filterFunc any) Validation
```

Comment:
Nah, it's not a good idea since the filter function can vary.
![image](https://github.com/goravel/framework/assets/84431594/02ee5655-b9ac-4519-b07c-8d189277f26d)

Reply:
Maybe:

```
type Filter interface {
	// Signature set the unique signature of the filter.
	Signature() string
	// Passes determine if the validation rule passes.
	Handle(val any) (any, error)
}
```

Reply:
> Nah, it's not a good idea since the filter function can vary.

We can only support one style: `func(val any) (any, error)`

Reply:
User can also create filter function like 
```go
func Default(val string, def ...string) string{
...
}
```
which can be used like 
```go
map[string]string{
"name" : "default:krishan"
}
```

Reply:
> Maybe:
> 
> ```
> type Filter interface {
> 	// Signature set the unique signature of the filter.
> 	Signature() string
> 	// Passes determine if the validation rule passes.
> 	Handle() func(val any) (any, error)
> }
> ```

we can change it to 
```go
type Filter interface {
	// Signature set the unique signature of the filter.
	Signature() string
	// Passes determine if the validation rule passes.
	Handle() any // will mention that any can only be two type of function
}
```

Reply:
We also need to add a new `Filter` method in the `FormRequest`:

<img width="928" alt="image" src="https://github.com/goravel/framework/assets/24771476/f8614526-16bb-4948-9e8c-7ab4bdbd969a">

Users can use the Filter feature through the `Filter` method, instead of set a tag in the struct:

<img width="537" alt="image" src="https://github.com/goravel/framework/assets/24771476/d0ad9066-63d4-4a24-a6c4-1deef30e1ac6">

The way of set a tag in the struct is inconsistent with the rule, so I suggest optimize it. cc @devhaozi 

So we need to fdmovd the logic in [goravel/gin](https://github.com/goravel/gin/pull/33/files#diff-c45c9f86e31f685d97279261d96ded9782dfe0473aa5666be11e9ffa14cbe46fR388) and [goravel/fiber](https://github.com/goravel/fiber/pull/38/files#diff-c45c9f86e31f685d97279261d96ded9782dfe0473aa5666be11e9ffa14cbe46fR413) and add some new logic:

<img width="1721" alt="image" src="https://github.com/goravel/framework/assets/24771476/fe57657d-027a-42ca-9f32-94a27afe99c2">